macros/graph/VectorField2D.pl

```diff
@@ -0,0 +1,113 @@
+sub _VectorField2D_init { };    # don't reload this file
+
+loadMacros("MathObjects.pl", "PGgraphmacros.pl");
+
+sub VectorField2D {
+
+###########################################
+	#
+	#  Set default options
+	#
+
+	my %options = (
+		graphobject     => $gr,
+		Fx              => Formula("1"),
+		Fy              => Formula("1"),
+		xvar            => 'x',
+		yvar            => 'y',
+		xmin            => -5,
+		xmax            => 5,
+		ymin            => -5,
+		ymax            => 5,
+		xsamples        => 10,
+		ysamples        => 10,
+		vectorcolor     => "blue",
+		vectorscale     => 0.25,
+		vectorthickness => 2,
+		vectortipwidth  => 0.08,
+		vectortiplength => 0.65,
+		xavoid          => 1000000,
+		yavoid          => 1000000,
+		@_
+	);
+
+	my $Fxsubroutine;
+	my $Fysubroutine;
+
+	$options{Fx}->perlFunction('Fxsubroutine', [ "$options{xvar}", "$options{yvar}" ]);
+	$options{Fy}->perlFunction('Fysubroutine', [ "$options{xvar}", "$options{yvar}" ]);
+
+######################################################
+	#
+	#  Generate plot data
+	#
+
+	my $dx = ($options{xmax} - $options{xmin}) / $options{xsamples};
+	my $dy = ($options{ymax} - $options{ymin}) / $options{ysamples};
+
+	my $xtail;
+	my $ytail;
+	my $FX;
+	my $FY;
+	my $xtip;
+	my $ytip;
+	my $xstem;
+	my $ystem;
+	my $xmidtip;
+	my $ymidtip;
+	my $xleftbarb;
+	my $xrightbarb;
+	my $yleftbarb;
+	my $yrightbarb;
+
+	foreach my $i (0 .. $options{xsamples}) {
```

Comment:
switch `foreach` to `for` here and below.

Reply:
This was just a copy of the previous macro, starting to clean up the way it is coded opens up a lot of other issues as well. As such I just refactored the whole macro to hopefully be a little more efficient. A bigger issue was the fact of storing the information about every arrow in a double array, which was never used.

Reply:
I guess we can just keep as is then.  On our todo list, we'd like to clean up all of these macros at some point.

Reply:
I already cleaned this one up and added some POD to it.